receiver/k8sclusterreceiver/internal/collection/nodes.go

```diff
@@ -61,27 +62,46 @@ func getMetricsForNode(node *corev1.Node, nodeConditionTypesToReport, allocatabl
 			},
 		})
 	}
+
 	// Adding 'node allocatable type' metrics
 	for _, nodeAllocatableTypeValue := range allocatableTypesToReport {
 		nodeAllocatableMetric := getNodeAllocatableMetric(nodeAllocatableTypeValue)
 		v1NodeAllocatableTypeValue := corev1.ResourceName(nodeAllocatableTypeValue)
-		metricValue, err := nodeAllocatableValue(node, v1NodeAllocatableTypeValue)
-
-		// metrics will be skipped if metric not present in node or value is not convertable to int64
-		if err != nil {
-			logger.Debug(err.Error())
+		valType := metricspb.MetricDescriptor_GAUGE_INT64
+		quantity, ok := node.Status.Allocatable[v1NodeAllocatableTypeValue]
+		if !ok {
+			logger.Debug(fmt.Errorf("allocatable type %v not found in node %v", nodeAllocatableTypeValue,
+				node.GetName()).Error())
+			continue
+		}
+		val := utils.GetInt64TimeSeries(quantity.Value())
+
+		if featuregate.IsEnabled(reportCPUMetricsAsDoubleFeatureGateID) {
+			// cpu metrics must be of the double type to adhere to opentelemetry system.cpu metric specifications
+			if v1NodeAllocatableTypeValue == corev1.ResourceCPU {
+				val = utils.GetDoubleTimeSeries(float64(quantity.MilliValue()) / 1000.0)
+				valType = metricspb.MetricDescriptor_GAUGE_DOUBLE
+			}
 		} else {
-			metrics = append(metrics, &metricspb.Metric{
-				MetricDescriptor: &metricspb.MetricDescriptor{
-					Name:        nodeAllocatableMetric,
-					Description: allocatableDesciption[v1NodeAllocatableTypeValue.String()],
-					Type:        metricspb.MetricDescriptor_GAUGE_INT64,
-				},
-				Timeseries: []*metricspb.TimeSeries{
-					utils.GetInt64TimeSeries(metricValue),
-				},
-			})
+			// metrics will be skipped if metric not present in node or value is not convertable to int64
+			valInt64, ok := quantity.AsInt64()
+			if !ok {
+				logger.Debug(fmt.Errorf("metric %s has value %v which is not convertable to int64",
+					v1NodeAllocatableTypeValue, node.GetName()).Error())
+				continue
+			}
+			val = utils.GetInt64TimeSeries(valInt64)
```

Comment:
I believe this part can be just removed, right? 

Feel free to remove it now or in a followup PR to keep this change set clean.

Reply:
Let's remove it later, I think we want to keep this for backwards compatibility. We get the quantity value with different methods that do similar things but are slightly different (quantity.Value() vs quantity.AsInt64()).

Reply:
> We get the quantity value with different methods that do similar things but are slightly different (quantity.Value() vs quantity.AsInt64()).

How different they are? Looking at the code seems like the value must be always the same.

Reply:
I was thinking the case where the quantity value is not an int value might be an issue. quantity.Value() will round a non-int value up to the nearest integer away from 0. quantity.AsInt64() will return false if if the quantity value is not an int 

Now that I think about it, with the conditional check the values should always be the same here. 

Will remove this in the future still.